aws/resource_aws_appautoscaling_scheduled_action.go

```diff
@@ -0,0 +1,222 @@
+
+func resourceAwsAppautoscalingScheduledAction() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppautoscalingScheduledActionCreate,
+		Read:   resourceAwsAppautoscalingScheduledActionRead,
+		Delete: resourceAwsAppautoscalingScheduledActionDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"service_namespace": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					validServiceNamespaces := []string{"ecs", "elasticmapreduce", "ec2", "appstream", "dynamodb"}
+					for _, str := range validServiceNamespaces {
+						if value == str {
+							return
+						}
+					}
+					errors = append(errors, fmt.Errorf("expected %s to be one of %v, got %s", k, validServiceNamespaces, value))
+					return
+				},
+			},
+			"resource_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"scalable_dimension": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					validDimensions := []string{"ecs:service:DesiredCount", "ec2:spot-fleet-request:TargetCapacity",
+						"elasticmapreduce:instancegroup:InstanceCount", "appstream:fleet:DesiredCapacity",
+						"dynamodb:table:ReadCapacityUnits", "dynamodb:table:WriteCapacityUnits",
+						"dynamodb:index:ReadCapacityUnits", "dynamodb:index:WriteCapacityUnits"}
+					for _, str := range validDimensions {
+						if value == str {
+							return
+						}
+					}
+					errors = append(errors, fmt.Errorf("expected %s to be one of %v, got %s", k, validDimensions, value))
+					return
+				},
+			},
+			"scalable_target_action": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_capacity": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: true,
+						},
+						"min_capacity": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"schedule": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"start_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"end_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppautoscalingScheduledActionCreate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Per [docs](http://docs.aws.amazon.com/cli/latest/reference/application-autoscaling/put-scheduled-action.html) I think we should support updates of this resource and basically just rename this function to something like `resourceAwsAppautoscalingScheduledActionCreateOrUpdate` or `resourceAwsAppautoscalingScheduledActionPut` and reference it in `Update` too.

Reply:
OK👍